coderd/apikey.go

```diff
@@ -338,35 +341,101 @@ func (api *API) deleteAPIKey(rw http.ResponseWriter, r *http.Request) {
 // @Success 200 {object} codersdk.TokenConfig
 // @Router /users/{user}/keys/tokens/tokenconfig [get]
 func (api *API) tokenConfig(rw http.ResponseWriter, r *http.Request) {
-	values, err := api.DeploymentValues.WithoutSecrets()
-	if err != nil {
-		httpapi.InternalServerError(rw, err)
-		return
+	ctx := r.Context()
+
+	maxLifetime := api.DeploymentValues.Sessions.MaximumTokenDuration.Value()
+
+	user := httpmw.UserParam(r)
+	if user.ID != uuid.Nil {
+		subject, userStatus, err := httpmw.UserRBACSubject(ctx, api.Database, user.ID, rbac.ScopeAll)
+		if err == nil && userStatus != database.UserStatusSuspended {
+			maxLifetime = api.getMaxTokenLifetimeForUser(ctx, subject)
+		}
+	} else {
+		api.Logger.Warn(ctx, "user ID is nil in token config request context")
 	}
```

Comment:
I would invert the condition and fail the request early in this case, although I don't see how the user ID could be nil here.